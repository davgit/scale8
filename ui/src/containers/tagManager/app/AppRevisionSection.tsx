import { FC } from 'react';
import { useQuery } from '@apollo/client';
import TagIcon from '../../../components/atoms/Icons/TagIcon';
import PlatformRevisionIcon from '../../../components/atoms/Icons/PlatformRevisionIcon';
import NavAppRevisionQuery from '../../../gql/queries/NavAppRevisionQuery';
import { NavAppRevision } from '../../../gql/generated/NavAppRevision';
import {
    buildAppButtonProps,
    buildAppRevisionButtonProps,
    buildOrgButtonProps,
    buildTagManagerButtonProps,
} from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import ActionIcon from '../../../components/atoms/Icons/ActionIcon';
import TriggerIcon from '../../../components/atoms/Icons/TriggerIcon';
import { buildAppRevisionBreadcrumbActions } from '../../../utils/BuildAppRevisionBreadcrumbActions';
import { extractPermissionsFromOrgUser } from '../../../context/OrgUserReducer';
import { SectionKey } from '../../SectionsDetails';
import { useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { toAppRevision } from '../../../utils/NavigationPaths';
import { analyticsEnabled } from '../../../utils/AnalyticsUtils';

const AppRevisionSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();
    const { ask, dispatchDialogAction, setRefreshCurrentPage } = templateInteractions;

    const currentOrgPermissions = extractPermissionsFromOrgUser(orgUserState);

    const sectionProps: SectionProps<NavAppRevision> = {
        children,
        sectionKey: SectionKey.appRevision,
        queryResult: useQuery<NavAppRevision>(NavAppRevisionQuery, {
            variables: { id },
        }),
        initContext: (data) => {
            templateInteractions.setSectionHasAnalytics(analyticsEnabled(data.getRevision.app));
        },
        buildButtonsProps: (data) => [
            buildOrgButtonProps(
                router,
                data.me.orgs,
                data.getRevision.app.tag_manager_account.org.id,
                data.getRevision.app.tag_manager_account.org.name,
            ),
            buildTagManagerButtonProps(
                router,
                data.getRevision.app.tag_manager_account.id,
                data.getRevision.app.tag_manager_account.org.data_manager_account?.id ?? '',
            ),
            buildAppButtonProps(
                router,
                data.getRevision.app.tag_manager_account.apps,
                data.getRevision.app.id,
                data.getRevision.app.name,
            ),
            buildAppRevisionButtonProps(
                router,
                data.getRevision.app.revisions,
                id,
                data.getRevision.name,
                true,
            ),
        ],
        buildMenuItemsProps: () => [
            {
                icon: () => <TagIcon />,
                label: 'Tags',
                link: toAppRevision({ id }, 'tags'),
            },
            {
                icon: () => <TriggerIcon />,
                label: 'Global Triggers',
                link: toAppRevision({ id }, 'global-triggers'),
            },
            {
                icon: () => <ActionIcon />,
                label: 'Global Actions',
                link: toAppRevision({ id }, 'global-actions'),
            },
            {
                icon: () => <PlatformRevisionIcon />,
                label: 'Platform Revisions',
                link: toAppRevision({ id }, 'app-platform-revisions'),
            },
        ],
        extractOrgUserDetails: (data) => data.getRevision.app.tag_manager_account.org,
        buildBreadcrumbActions: (data) =>
            buildAppRevisionBreadcrumbActions(
                id,
                data.getRevision.name,
                router,
                currentOrgPermissions,
                dispatchDialogAction,
                setRefreshCurrentPage,
                ask,
                data.getRevision.locked,
            ),
        accountExpireIn: orgUserState?.tagManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavAppRevision> {...sectionProps} />;
};

export default AppRevisionSection;
