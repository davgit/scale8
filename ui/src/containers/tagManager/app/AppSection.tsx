import { FC } from 'react';
import { useQuery } from '@apollo/client';
import { NavApp } from '../../../gql/generated/NavApp';
import NavAppQuery from '../../../gql/queries/NavAppQuery';
import RevisionIcon from '../../../components/atoms/Icons/RevisionIcon';
import EnvironmentIcon from '../../../components/atoms/Icons/EnvironmentIcon';
import PlatformIcon from '../../../components/atoms/Icons/PlatformIcon';
import {
    buildAppButtonProps,
    buildOrgButtonProps,
    buildTagManagerButtonProps,
} from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import AppAnalyticsIcon from '../../../components/atoms/Icons/AppAnalyticsIcon';
import { SectionKey } from '../../SectionsDetails';
import { useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { toApp } from '../../../utils/NavigationPaths';
import { analyticsEnabled } from '../../../utils/AnalyticsUtils';
import { SideMenuButtonProps } from '../../../components/molecules/SideMenuButton';

const AppSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();

    const sectionProps: SectionProps<NavApp> = {
        children,
        sectionKey: SectionKey.app,
        queryResult: useQuery<NavApp>(NavAppQuery, {
            variables: { id },
        }),
        initContext: (data) => {
            templateInteractions.setSectionHasAnalytics(analyticsEnabled(data.getApp));
        },
        buildButtonsProps: (data) => [
            buildOrgButtonProps(
                router,
                data.me.orgs,
                data.getApp.tag_manager_account.org.id,
                data.getApp.tag_manager_account.org.name,
            ),
            buildTagManagerButtonProps(
                router,
                data.getApp.tag_manager_account.id,
                data.getApp.tag_manager_account.org.data_manager_account?.id ?? '',
            ),
            buildAppButtonProps(
                router,
                data.getApp.tag_manager_account.apps,
                id,
                data.getApp.name,
                true,
            ),
        ],
        buildMenuItemsProps: (_, data) => [
            ...((analyticsEnabled(data.getApp)
                ? [
                      {
                          icon: <AppAnalyticsIcon />,
                          label: 'Analytics',
                          link: toApp({ id }, 'analytics'),
                      },
                  ]
                : []) as SideMenuButtonProps[]),
            {
                icon: <RevisionIcon />,
                label: 'Revisions',
                link: toApp({ id }, 'revisions'),
            },
            {
                icon: <EnvironmentIcon />,
                label: 'Environments',
                link: toApp({ id }, 'environments'),
            },
            {
                icon: <PlatformIcon />,
                label: 'Installed Platforms',
                link: toApp({ id }, 'platforms'),
            },
        ],
        extractOrgUserDetails: (data) => data.getApp.tag_manager_account.org,
        accountExpireIn: orgUserState?.tagManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavApp> {...sectionProps} />;
};

export default AppSection;
