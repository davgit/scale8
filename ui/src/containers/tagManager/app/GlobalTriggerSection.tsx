import { FC } from 'react';
import { useQuery } from '@apollo/client';
import NavGlobalTriggerQuery from '../../../gql/queries/NavGlobalTriggerQuery';
import { NavGlobalTrigger } from '../../../gql/generated/NavGlobalTrigger';
import {
    buildAppButtonProps,
    buildAppRevisionButtonProps,
    buildGlobalTriggerButtonProps,
    buildOrgButtonProps,
    buildTagManagerButtonProps,
} from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import { buildAppRevisionBreadcrumbActions } from '../../../utils/BuildAppRevisionBreadcrumbActions';
import { extractPermissionsFromOrgUser } from '../../../context/OrgUserReducer';
import { SectionKey } from '../../SectionsDetails';
import { useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { analyticsEnabled } from '../../../utils/AnalyticsUtils';

const GlobalTriggerSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();

    const { ask, dispatchDialogAction, setRefreshCurrentPage } =
        templateInteractions;
    const currentOrgPermissions = extractPermissionsFromOrgUser(orgUserState);

    const sectionProps: SectionProps<NavGlobalTrigger> = {
        children,
        sectionKey: SectionKey.globalTrigger,
        queryResult: useQuery<NavGlobalTrigger>(NavGlobalTriggerQuery, {
            variables: { id },
        }),
        initContext: (data) => {
            templateInteractions.setSectionHasAnalytics(
                analyticsEnabled(data.getTrigger.revision.app),
            );
        },
        buildButtonsProps: (data) => [
            buildOrgButtonProps(
                router,
                data.me.orgs,
                data.getTrigger.revision.app.tag_manager_account.org.id,
                data.getTrigger.revision.app.tag_manager_account.org.name,
            ),
            buildTagManagerButtonProps(
                router,
                data.getTrigger.revision.app.tag_manager_account.id,
                data.getTrigger.revision.app.tag_manager_account.org.data_manager_account?.id ?? '',
            ),
            buildAppButtonProps(
                router,
                data.getTrigger.revision.app.tag_manager_account.apps,
                data.getTrigger.revision.app.id,
                data.getTrigger.revision.app.name,
            ),
            buildAppRevisionButtonProps(
                router,
                data.getTrigger.revision.app.revisions,
                data.getTrigger.revision.id,
                data.getTrigger.revision.name,
            ),
            buildGlobalTriggerButtonProps(
                router,
                data.getTrigger.revision.global_triggers,
                id,
                data.getTrigger.name,
                true,
            ),
        ],
        buildMenuItemsProps: () => [],
        extractOrgUserDetails: (data) => data.getTrigger.revision.app.tag_manager_account.org,
        buildBreadcrumbActions: (data) =>
            buildAppRevisionBreadcrumbActions(
                data.getTrigger.revision.id,
                data.getTrigger.revision.name,
                router,
                currentOrgPermissions,
                dispatchDialogAction,
                setRefreshCurrentPage,
                ask,
                data.getTrigger.revision.locked,
            ),
        accountExpireIn: orgUserState?.tagManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavGlobalTrigger> {...sectionProps} />;
};

export default GlobalTriggerSection;
