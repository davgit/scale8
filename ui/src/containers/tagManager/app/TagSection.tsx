import { FC } from 'react';
import { useQuery } from '@apollo/client';
import NavTagQuery from '../../../gql/queries/NavTagQuery';
import { NavTag } from '../../../gql/generated/NavTag';
import {
    buildAppButtonProps,
    buildAppRevisionButtonProps,
    buildOrgButtonProps,
    buildTagButtonProps,
    buildTagManagerButtonProps,
} from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import { buildAppRevisionBreadcrumbActions } from '../../../utils/BuildAppRevisionBreadcrumbActions';
import { extractPermissionsFromOrgUser } from '../../../context/OrgUserReducer';
import { SectionKey } from '../../SectionsDetails';
import { useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { analyticsEnabled } from '../../../utils/AnalyticsUtils';

const TagSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();

    const { ask, dispatchDialogAction, setRefreshCurrentPage } =
        templateInteractions;
    const currentOrgPermissions = extractPermissionsFromOrgUser(orgUserState);

    const sectionProps: SectionProps<NavTag> = {
        children,
        sectionKey: SectionKey.tag,
        queryResult: useQuery<NavTag>(NavTagQuery, {
            variables: { id },
        }),
        initContext: (data) => {
            templateInteractions.setSectionHasAnalytics(analyticsEnabled(data.getTag.revision.app));
        },
        buildButtonsProps: (data) => [
            buildOrgButtonProps(
                router,
                data.me.orgs,
                data.getTag.revision.app.tag_manager_account.org.id,
                data.getTag.revision.app.tag_manager_account.org.name,
            ),
            buildTagManagerButtonProps(
                router,
                data.getTag.revision.app.tag_manager_account.id,
                data.getTag.revision.app.tag_manager_account.org.data_manager_account?.id ?? '',
            ),
            buildAppButtonProps(
                router,
                data.getTag.revision.app.tag_manager_account.apps,
                data.getTag.revision.app.id,
                data.getTag.revision.app.name,
            ),
            buildAppRevisionButtonProps(
                router,
                data.getTag.revision.app.revisions,
                data.getTag.revision.id,
                data.getTag.revision.name,
            ),
            buildTagButtonProps(router, data.getTag.revision.tags, id, data.getTag.name, true),
        ],
        buildMenuItemsProps: () => [],
        extractOrgUserDetails: (data) => data.getTag.revision.app.tag_manager_account.org,
        buildBreadcrumbActions: (data) =>
            buildAppRevisionBreadcrumbActions(
                data.getTag.revision.id,
                data.getTag.revision.name,
                router,
                currentOrgPermissions,
                dispatchDialogAction,
                setRefreshCurrentPage,
                ask,
                data.getTag.revision.locked,
            ),
        accountExpireIn: orgUserState?.tagManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavTag> {...sectionProps} />;
};

export default TagSection;
