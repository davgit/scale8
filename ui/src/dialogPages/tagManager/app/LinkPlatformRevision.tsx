import { FC } from 'react';
import { ApolloError, useMutation, useQuery } from '@apollo/client';
import { FetchAvailableAppPlatformRevisions } from '../../../gql/generated/FetchAvailableAppPlatformRevisions';
import FetchAvailableAppPlatformRevisionsQuery from '../../../gql/queries/FetchAvailableAppPlatformRevisionsQuery';
import AppPlatformRevisionForm from '../../../components/organisms/Forms/AppPlatformRevisionForm';
import LinkAppPlatformRevisionQuery from '../../../gql/mutations/LinkAppPlatformRevisionQuery';
import { LinkAppPlatformRevisionValues } from '../../../gql/generated/LinkAppPlatformRevisionValues';
import {
    FormWithMappedPlatformValuesResult,
    ModelWithPlatformDataMaps,
} from '../../../hooks/form/useFormWithMappedPlatformValues';
import { DataMap, PlatformDataMap } from '../../../types/DataMapsTypes';
import {
    MappedPlatformValues,
    ValuesWithMappedPlatformData,
} from '../../../types/MappedPlatformValuesTypes';
import { AppPlatformRevisionLinkInput, DataMapInput } from '../../../gql/generated/globalTypes';
import { mappedPlatformValuesToDataMapInput } from '../../../utils/MappedPlatformValuesUtils';
import { DialogPageProps } from '../../../types/DialogTypes';
import { buildStandardFormInfo } from '../../../utils/InfoLabelsUtils';
import { FormBaseProps } from '../../../types/props/forms/CommonFormProps';
import { DialogPreloadForm, DialogPreloadFormProps } from '../../abstractions/DialogPreloadForm';

export type ConnectPlatformRevisionAfterLoadLoadProps = DialogPageProps & {
    availableAppPlatformRevisions: {
        key: string;
        text: string;
        sub?: { key: string; text: string }[];
    }[];
    appPlatformRevisionsWithSettings: ModelWithPlatformDataMaps[];
    isUpdate: boolean;
    currentPlatformRevisionId?: string;
    currentPlatformSettings?: DataMap[];
};

export type AppPlatformRevisionBaseValues = {
    platformRevisionId: string;
};

export type AppPlatformRevisionValues = ValuesWithMappedPlatformData<AppPlatformRevisionBaseValues>;

export type AppPlatformRevisionFormProps =
    FormWithMappedPlatformValuesResult<AppPlatformRevisionValues> &
        FormBaseProps & {
            isEdit: boolean;
            initialId: string;
            availableAppPlatformRevisions: {
                key: string;
                text: string;
                sub?: { key: string; text: string }[];
            }[];
        };

export const buildAppPlatformRevisionLinkInput = (
    revisionId: string,
    platformRevisionId: string,
    mappedPlatformValues: MappedPlatformValues,
    preview?: boolean,
): AppPlatformRevisionLinkInput => {
    const data_maps: DataMapInput[] = mappedPlatformValuesToDataMapInput(mappedPlatformValues);

    return {
        revision_id: revisionId,
        platform_revision_id: platformRevisionId,
        data_maps,
        preview,
    };
};

const LinkPlatformRevision: FC<DialogPageProps> = (props: DialogPageProps) => {
    const revisionId = props.id;

    const linkPlatformRevisionProps: DialogPreloadFormProps<
        FetchAvailableAppPlatformRevisions,
        AppPlatformRevisionValues,
        AppPlatformRevisionFormProps,
        LinkAppPlatformRevisionValues
    > = {
        loadQuery: useQuery<FetchAvailableAppPlatformRevisions>(
            FetchAvailableAppPlatformRevisionsQuery,
            {
                variables: { id: revisionId },
            },
        ),
        buildInitialStatePreload: () => ({
            platformRevisionId: '',
        }),
        saveQuery: useMutation<LinkAppPlatformRevisionValues>(LinkAppPlatformRevisionQuery),
        mapSaveData: (appPlatformRevisionValues: AppPlatformRevisionValues) => {
            const appPlatformRevisionLinkInput = buildAppPlatformRevisionLinkInput(
                revisionId,
                appPlatformRevisionValues.platformRevisionId,
                appPlatformRevisionValues.mappedPlatformValues ?? [],
            );
            return { appPlatformRevisionLinkInput };
        },
        buildFormProps: (
            formLoadedData: FetchAvailableAppPlatformRevisions,
            formValidationValues: FormWithMappedPlatformValuesResult<AppPlatformRevisionValues>,
            gqlError?: ApolloError,
        ) => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Connect Platform Revision',
            title: 'Connect Platform Revision',
            formInfoProps: buildStandardFormInfo('appPlatformRevisions', 'Link'),
            handleDialogClose: props.handleDialogClose,
            isEdit: false,
            initialId: '',
            availableAppPlatformRevisions: formLoadedData.getRevision.app.app_platforms
                .filter(
                    (_) =>
                        !formLoadedData.getRevision.app_platform_revisions
                            .map((p) => p.platform_revision.platform.id)
                            .includes(_.platform.id),
                )
                .map((_) => ({
                    key: _.platform.id,
                    text: _.platform.name,
                    sub: _.platform.platform_revisions
                        .filter((_) => _.locked)
                        .map((r) => ({
                            key: r.id,
                            text: r.name,
                        })),
                })),
        }),
        checkSuccessfullySubmitted: (formMutationData) => {
            return formMutationData?.linkPlatformRevision !== undefined;
        },
        pageComponent: AppPlatformRevisionForm,
        validators: [],
        mappedPlatformValuesProps: {
            buildAvailableModelsWithPlatformDataMaps: (
                formLoadedData: FetchAvailableAppPlatformRevisions,
            ) => {
                const filteredPlatforms = formLoadedData.getRevision.app.app_platforms.filter(
                    (_) =>
                        !formLoadedData.getRevision.app_platform_revisions
                            .map((p) => p.platform_revision.platform.id)
                            .includes(_.platform.id),
                );
                return filteredPlatforms
                    .map((_) =>
                        _.platform.platform_revisions.map((r) => ({
                            id: r.id,
                            platformDataMaps: r.platform_settings as PlatformDataMap[],
                        })),
                    )
                    .flat();
            },
            idValueForModelWithPlatformDataMaps: 'platformRevisionId',
        },
        ...props,
    };

    return (
        <DialogPreloadForm<
            FetchAvailableAppPlatformRevisions,
            AppPlatformRevisionValues,
            AppPlatformRevisionFormProps,
            LinkAppPlatformRevisionValues
        >
            {...linkPlatformRevisionProps}
        />
    );
};

export default LinkPlatformRevision;
