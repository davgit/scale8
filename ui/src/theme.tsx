import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
