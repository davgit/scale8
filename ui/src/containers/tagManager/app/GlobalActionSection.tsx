import { FC } from 'react';
import { useQuery } from '@apollo/client';
import { buildGlobalActionButtonProps } from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import NavGlobalActionQuery from '../../../gql/queries/NavGlobalActionQuery';
import { NavGlobalAction } from '../../../gql/generated/NavGlobalAction';
import { buildAppRevisionBreadcrumbActions } from '../../../utils/BuildAppRevisionBreadcrumbActions';
import { extractPermissionsFromOrgUser } from '../../../context/OrgUserReducer';
import { SectionKey } from '../../SectionsDetails';
import { useConfigState, useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { analyticsEnabled, errorTrackingEnabled } from '../../../utils/AnalyticsUtils';
import { buildAppRevisionButtons } from './AppRevisionSection';

const GlobalActionSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();
    const { useSignup } = useConfigState();

    const { ask, dispatchDialogAction, setRefreshCurrentPage } = templateInteractions;
    const currentOrgPermissions = extractPermissionsFromOrgUser(orgUserState);

    const sectionProps: SectionProps<NavGlobalAction> = {
        children,
        sectionKey: SectionKey.globalAction,
        queryResult: useQuery<NavGlobalAction>(NavGlobalActionQuery, {
            variables: { id },
        }),
        sectionHasAnalytics: (data) =>
            analyticsEnabled(data.getActionGroupDistribution.revision.app),
        buildButtonsProps: (data, orgPermissions) => [
            ...buildAppRevisionButtons(
                data.me.orgs,
                data.getActionGroupDistribution.revision.app.tag_manager_account.org,
                data.getActionGroupDistribution.revision.app.tag_manager_account,
                data.getActionGroupDistribution.revision.app.tag_manager_account.org
                    .data_manager_account,
                data.getActionGroupDistribution.revision.app.tag_manager_account.apps,
                data.getActionGroupDistribution.revision.app,
                data.getActionGroupDistribution.revision.app.revisions,
                data.getActionGroupDistribution.revision,
                analyticsEnabled(data.getActionGroupDistribution.revision.app),
                errorTrackingEnabled(data.getActionGroupDistribution.revision.app),
                router,
                orgPermissions,
                useSignup,
                'Global Actions',
            ),
            buildGlobalActionButtonProps(
                router,
                data.getActionGroupDistribution.revision.global_action_group_distributions,
                id,
                data.getActionGroupDistribution.name,
                true,
            ),
        ],
        buildMenuItemsProps: () => [],
        extractOrgUserDetails: (data) =>
            data.getActionGroupDistribution.revision.app.tag_manager_account.org,
        buildBreadcrumbActions: (data) =>
            buildAppRevisionBreadcrumbActions(
                data.getActionGroupDistribution.revision.id,
                data.getActionGroupDistribution.revision.name,
                router,
                currentOrgPermissions,
                dispatchDialogAction,
                setRefreshCurrentPage,
                ask,
                data.getActionGroupDistribution.revision.locked,
            ),
        accountExpireIn: orgUserState?.tagManagerAccount.trialExpiration ?? undefined,
        accountExpired: orgUserState?.tagManagerAccount.trialExpired ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount.isTrial ?? undefined,
    };

    return <Section<NavGlobalAction> {...sectionProps} />;
};

export default GlobalActionSection;
