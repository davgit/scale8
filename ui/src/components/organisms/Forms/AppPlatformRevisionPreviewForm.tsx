import { FC, Fragment } from 'react';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import { FormBaseProps } from '../../../types/props/forms/CommonFormProps';
import { FormValidationResult } from '../../../hooks/form/useFormValidation';
import { DialogContentText, List, ListItem, ListItemText } from '@mui/material';
import { AppPlatformRevisionValues } from '../../../dialogPages/tagManager/app/LinkPlatformRevision';

export type AppPlatformRevisionPreviewProps = FormValidationResult<AppPlatformRevisionValues> &
    FormBaseProps & {
        revisionName: string;
        groups: { title: string; items: { name: string; id: string }[] }[];
    };

const AppPlatformRevisionPreviewForm: FC<AppPlatformRevisionPreviewProps> = (
    props: AppPlatformRevisionPreviewProps,
) => {
    return (
        <DrawerFormLayout {...props}>
            <DialogContentText id="alert-dialog-description">
                {props.groups.length === 0
                    ? `Change to Revision: ${props.revisionName}?`
                    : `Changing to Revision "${props.revisionName}" will require the removal of:`}
            </DialogContentText>
            <List
                dense
                sx={{
                    padding: (theme) => theme.spacing(0, 2),
                }}
            >
                {props.groups.map((group, key) => (
                    <Fragment key={key}>
                        <ListItemText secondary={group.title} />
                        {group.items.map((item, key) => (
                            <ListItem key={key}>
                                <ListItemText primary={`${item.name} (${item.id})`} />
                            </ListItem>
                        ))}
                    </Fragment>
                ))}
            </List>
        </DrawerFormLayout>
    );
};

export default AppPlatformRevisionPreviewForm;
