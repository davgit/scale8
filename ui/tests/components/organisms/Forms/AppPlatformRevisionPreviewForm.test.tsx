import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import AppPlatformRevisionPreviewForm from '../../../../src/components/organisms/Forms/AppPlatformRevisionPreviewForm';
import { AppPlatformRevisionValues } from '../../../../src/dialogPages/tagManager/app/LinkPlatformRevision';

const MockForm: FC = () => {
    const formProps = useMockFormProps<AppPlatformRevisionValues>({
        platformRevisionId: '',
    });

    return <AppPlatformRevisionPreviewForm {...formProps} revisionName="" groups={[]} />;
};

describe('AppPlatformRevisionForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
