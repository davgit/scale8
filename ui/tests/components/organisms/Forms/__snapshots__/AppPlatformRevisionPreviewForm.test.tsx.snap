// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPlatformRevisionForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        mock form
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <p
            class="MuiDialogContentText-root MuiTypography-root MuiTypography-body1 css-s2fmnb-MuiTypography-root-MuiDialogContentText-root"
            id="alert-dialog-description"
          >
            Change to Revision: ?
          </p>
          <ul
            class="MuiList-root MuiList-padding MuiList-dense css-160lz9-MuiList-root"
          />
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
